### PR TITLE
Add warning if there are remote artifacts with no allowed checksums

### PR DESCRIPTION
**Pull request by daviddavis**
fixes #7985

Please be sure you have read our documentation on creating PRs:
https://docs.pulpproject.org/contributing/pull-request-walkthrough.html

### CODE REVIEW

File: pulpcore/app/settings.py [hunk 1]
```diff
@@ -341,6 +344,24 @@
                         ).format(checksum)
                     )
 
+            # warn if there are remote artifacts with checksums but no allowed checksums
+            cond = " AND ".join([f"{c} IS NULL" for c in constants.ALL_KNOWN_CONTENT_CHECKSUMS])
+            no_checksum_query = f"SELECT pulp_id FROM core_remoteartifact WHERE {cond}"
+            cond = " AND ".join([f"{c} IS NULL" for c in ALLOWED_CONTENT_CHECKSUMS])
```
Comment (ipanova) [inline pulpcore/app/settings.py:350]: does it make sense to make this query smaller and query only those remote artifacts which have no checksums from `ALLOWED_CONTENT_CHECKSUMS` list or am i missing some corner case?

Comment (daviddavis) [reply pulpcore/app/settings.py:350]: This is what the main part of the query does. However, we have to also exclude remote artifacts at all which is what the subquery is doing.

Comment (ipanova) [reply pulpcore/app/settings.py:350]: `RemoteArtifacts without any checksums which is perfectly valid and not prohibited.` yeah this is the part  i forgot about, thanks for reminding me

**Review by ipanova — APPROVED** [review]

**Review by bmbouter — APPROVED** [review]
This looks good.

**Pull request merged**